SDWebImage/SDImageAPNGCoder.m

```diff
@@ -199,10 +200,10 @@ - (NSData *)encodedDataWithImage:(UIImage *)image format:(SDImageFormat)format o
     }
     NSMutableDictionary *properties = [NSMutableDictionary dictionary];
     double compressionQuality = 1;
-    if ([options valueForKey:SDImageCoderEncodeCompressionQuality]) {
-        compressionQuality = [[options valueForKey:SDImageCoderEncodeCompressionQuality] doubleValue];
+    if (options[SDImageCoderEncodeCompressionQuality]) {
+        compressionQuality = [options[SDImageCoderEncodeCompressionQuality] doubleValue];
     }
-    [properties setValue:@(compressionQuality) forKey:(__bridge NSString *)kCGImageDestinationLossyCompressionQuality];
```

Comment:
Any reason to change this ? Because using `setObject:forKey:` on NSDictionary is not a good coding style.

`[NSDictionary setValue:forKey:]` will not throw exception and cause crash when the value is nil. However, `setObject:forKey:` does.

Reply:
Coding style and don't need to check `nil` in all the changes I make, I definitely sure they are not `nil`.

The code which has possible be nil, I still keep the `setValue:forKey`, like [these](https://github.com/rs/SDWebImage/blob/fe2fede60f9c6efcadfac8b90e8cae579656fffb/SDWebImage/SDWebImageManager.m#L372-L384).

Reply:
@dreampiggy Any consideration about this `PR`?